Convert ast-view to typescript

jeremy-rifkin opened the pull request:
Moves the ast view pane to TS

<!-- THIS COMMENT IS INVISIBLE IN THE FINAL PR, BUT FEEL FREE TO REMOVE IT
Thanks for taking the time to improve CE. We really appreciate it.
  Before opening the PR, please make sure that the tests & linter pass their checks,
  by running `make check`.
  In the best case scenario, you are also adding tests to back up your changes,
  but don't sweat it if you don't. We can discuss them at a later date.
Feel free to append your name to the CONTRIBUTORS.md file
Thanks again, we really appreciate this!
-->

jeremy-rifkin commented on static/panes/ast-view.ts:
```diff
@@ -0,0 +1,284 @@
+// Copyright (c) 2017, Simon Brand
```
Quick question: Should this line be updated or left with the original copyright from ast-view.js?

mattgodbolt replied:
I think it should be left alone. We can ask @TartanLlama if they'd like us to update the attribution, but for best I'd leave the copyright alone wtihout clear intent from the original author. Thanks!

RubenRBS commented on static/panes/ast-view.ts:
```diff
@@ -0,0 +1,284 @@
+
+'use strict';
+
+import _ from 'underscore';
```
Can you try to drop the underscore dependency? TS has support for most of the things that we need from _

RubenRBS commented on static/panes/ast-view.ts:
```diff
@@ -0,0 +1,284 @@
+'use strict';
+
+import _ from 'underscore';
+import $ from 'jquery';
```
This is already globally defined and can be dropped

RubenRBS commented on static/panes/ast-view.ts:
```diff
@@ -0,0 +1,284 @@
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+'use strict';
```
This is set by default when compiling ts code so no need to specify it and can be dropped

RubenRBS commented on static/panes/ast-view.ts:
```diff
@@ -0,0 +1,284 @@
+    }
+
+    onMouseMove(e: monaco.editor.IEditorMouseEvent) {
+        if (e === null || e.target === null || e.target.position === null) return;
```
```suggestion
        if (e?.target?.position === null) return;
```

RubenRBS reviewed (commented):
Thank you for taking the time to port this pane :)

jeremy-rifkin commented:
Alright, I've implemented the feedback. I removed most uses of underscore.js but it is still needed at least for `_.throttle`. (The remaining use of underscore.js is consistent ir-view.ts which I've been using as a reference for the conversion).

matsjla reviewed (approved):
LGTM - There is some duplicated code with the color/pane linking (with ir-view.ts), but I'll figure out a nice way to deal with that - Thanks!
